Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -230,14 +257,297 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateData()
         break;
     } // switch squareCase
     progress.CompletedPixel();
-  } // iteration
+  } // pixel square iteration
 
   // Now create the outputs paths from the deques we've been using.
   this->FillOutputs();
-  m_Contours.clear();
-  m_ContourStarts.clear();
-  m_ContourEnds.clear();
-  m_NumberOfContoursCreated = 0;
+}
+
+
+template <typename TInputImage>
+void
+ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels()
+{
+  using IndexType = typename InputRegionType::IndexType;
+  using RadiusType = typename ConstShapedNeighborhoodIterator<InputImageType>::RadiusType;
+  using RegionConstIterator = ImageRegionConstIterator<InputImageType>;
+  using RegionIterator = ImageRegionIterator<InputImageType>;
+  using SizeType = typename InputRegionType::SizeType;
+  using SquareIterator = ConstShapedNeighborhoodIterator<InputImageType>;
+
+  // Requested region
+  const InputRegionType inputRegion{ this->GetInput()->GetRequestedRegion() };
+
+  // Find all the distinct labels in the input region.
+  std::vector<InputPixelType> allLabels;
+  {
+    allLabels.reserve(inputRegion.GetNumberOfPixels());
+    RegionConstIterator inIt{ this->GetInput(), inputRegion };
+    for (inIt.GoToBegin(); !inIt.IsAtEnd(); ++inIt)
+    {
+      allLabels.push_back(inIt.Get());
+    }
+    std::sort(allLabels.begin(), allLabels.end());
+    const typename std::vector<InputPixelType>::const_iterator last{ std::unique(allLabels.begin(), allLabels.end()) };
+    allLabels.erase(last, allLabels.end());
+  }
+
+  // Make room so that largerImage and its largerRegion can have a one-pixel wide border on all sides of the input
+  // region.
+  const InputImagePointer largerImage{ InputImageType::New() };
+  const IndexType         largerIndex{ inputRegion.GetIndex()[0] - 1, inputRegion.GetIndex()[1] - 1 };
+  const SizeType          largerSize{ inputRegion.GetSize()[0] + 2, inputRegion.GetSize()[1] + 2 };
+  const InputRegionType   largerRegion{ largerIndex, largerSize };
+  largerImage->SetRegions(largerRegion);
+  largerImage->Allocate();
+  {
+    RegionConstIterator inputIt{ this->GetInput(), inputRegion };
+    RegionIterator      largerIt{ largerImage, inputRegion };
+    for (inputIt.GoToBegin(), largerIt.GoToBegin(); !inputIt.IsAtEnd(); ++inputIt, ++largerIt)
```

Comment:
When a local `inputImageRegionRange` variable is defined as in my previous comment, I think the six lines of code, starting with the declaration of `inputIt` may be replaced by the following two lines:

    ImageRegionRange<InputImageType> largerImageRegionRange{ *largerImage, inputRegion };
    std::copy(inputImageRegionRange.cbegin(), inputImageRegionRange.cend(), largerImageRegionRange.begin());

Reply:
Yes; see https://github.com/InsightSoftwareConsortium/ITK/pull/2323#discussion_r583050406.

Reply:
Yes to `std::copy`.

Reply:
And we can use
```c++
std::fill(stripeRange.begin(), stripeRange.end(), differentLabel);
```
... instead of a `for` loop in several places.  These `ImageRegionRange`s are handy!